### PR TITLE
JENKINS-247 Run SensorBox Tests inside docker container

### PR DESCRIPTION
**Pull request by eitzo**
Run the Nexus hardware tests inside a docker container.

### CODE REVIEW

File: Jenkinsfile.SensorboxTests [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+            // docker build directory avoids that overhead
+            dir 'docker'
+            // run on slave1 where nexus is connected to
+            label 'DeviceHardware'
+            // Pass the uid and the gid of the current user (jenkins-user) to the Dockerfile, so a
+            // corresponding user can be added. This is needed to provide the jenkins user inside
+            // the container for the ssh-agent to work.
+            // Another way would be to simply map the passwd file, but would spoil additional information
+            additionalBuildArgs '--build-arg USER_ID=$(id -u) --build-arg GROUP_ID=$(id -g)'
+            args "--net host -v /dev/bus/usb/:/dev/bus/usb --device /dev/kvm:/dev/kvm -v /var/local/container_shared/gradle/:/.gradle -v /var/local/container_shared/android-sdk:/usr/local/android-sdk -v /var/local/container_shared/android-home:/.android -v /var/local/container_shared/emulator_console_auth_token:/.emulator_console_auth_token -v /var/local/container_shared/analytics.settings:/analytics.settings"
```
Comment (mfuchs) [inline Jenkinsfile.SensorboxTests]: /dev/kvm is used by QEMU for the Android Emulator. So it shouldn't be necessary here.

File: Jenkinsfile.SensorboxTests [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
-              }
-         }
+     agent {
+        dockerfile {
+            filename 'Dockerfile.jenkins'
+            // 'docker build' would normally copy the whole build-dir to the container, changing the
+            // docker build directory avoids that overhead
+            dir 'docker'
+            // run on slave1 where nexus is connected to
+            label 'DeviceHardware'
```
Comment (mfuchs) [inline Jenkinsfile.SensorboxTests]: The test run here are specific to the sensor box. Thus it should be mandatory.
In the future more devices, including those with robot arm, will be added.
Thus I think we should be specific here and require the HardwareSensorBox.

File: Jenkinsfile.SensorboxTests [hunk 3]
```diff
@@ -1,17 +1,50 @@
...
+        ANDROID_NDK = ""
+        // This is important, as we want the keep our gradle cache, but we can't share it between containers
+        // the cache could only be shared if the gradle instances could comunicate with each other
+        // imho keeping the cache per executor will have the least space impact
+        GRADLE_USER_HOME = "/.gradle/${env.EXECUTOR_NUMBER}"
+        // Otherwise user.home returns ? for java applications
+        JAVA_TOOL_OPTIONS = "-Duser.home=/tmp/"
+
+        //// jenkins-android-helper related variables
+        // set to any value to debug jenkins_android* scripts
```
Comment (mfuchs) [inline Jenkinsfile.SensorboxTests]: This block, including ANDROID_EMULATOR_HELPER_DEBUG and PYTHONUNBUFFERED are not needed anymore.

Comment (eitzo) [reply Jenkinsfile.SensorboxTests]: Seems fine to me with the changes. 
See https://jenkins.catrob.at/job/lab/job/Sandro_Eitzinger/job/SensorBox-in-Docker/67/

**Pull request merged**